src/gsCore/gsMultiBasis.hpp

```diff
@@ -249,10 +249,91 @@ void gsMultiBasis<T>::uniformCoarsen_withTransfer(
     );
 
     // restrict to free dofs
-    combineTransferMatrices( localTransferMatrices, coarseMapper, fineMapper, transferMatrix );        
+    combineTransferMatrices( localTransferMatrices, coarseMapper, fineMapper, transferMatrix );
 
 }
 
+template <typename T>
+typename gsBasis<T>::uPtr gsMultiBasis<T>::componentBasis_withIndices(
+        patchComponent pc,
+        const gsDofMapper& dm,
+        gsMatrix<unsigned>& indices,
+        bool no_lower
+    ) const
+{
+    typename gsBasis<T>::uPtr result = m_bases[pc.patch()]->componentBasis_withIndices(pc, indices, no_lower);
+
+    const index_t sz = indices.rows();
+    index_t j = 0;
+    for (index_t i=0; i<sz; ++i)
+    {
+        const index_t loc = indices(i,0);
+        if (dm.is_free(loc, pc.patch()))
+        {
+            indices(j,0) = dm.index(loc, pc.patch());
+            ++j;
+        }
+    }
+    // Eigen does delete all data on resize. Thus, we have to do it on
+    // our own. We consider two important special cases: j==0 and j==sz.
+    if (j==0)
```

Comment:
use:
indices.conservativeResize(j,1);
so that Eigen does not delete all data.

Reply:
Oh, thanks.